src/api/libcellml/validator.h

```diff
@@ -49,10 +49,14 @@ class LIBCELLML_EXPORT Validator: public Logger
 private:
 
     void validateComponent(const ComponentPtr &component);
-    void validateUnits(const UnitsPtr &units, const std::vector<std::string> unitsNames);
+    void validateUnits(const UnitsPtr &units, const std::vector<std::string> &unitsNames);
+    void validateUnitsUnit(size_t index, const UnitsPtr &units, const std::vector<std::string> &unitsNames);
     void validateVariable(const VariablePtr &variable, std::vector<std::string> &variableNames);
     bool catchDoubleConversionError(const std::string &input);
 
+    bool isStandardUnitName(const std::string &name);
+    bool isStandardPrefixName(const std::string &name);
```

Comment:
All of these private methods above should be implemented in the private implementation so that they are not visible here in the API.